src/core/Menu/MenuItem.tsx

```diff
@@ -147,7 +147,7 @@ export const MenuItem = React.forwardRef<HTMLLIElement, MenuItemProps>(
         ref={refs}
         style={style}
         role={role}
-        tabIndex={disabled ? undefined : -1}
+        tabIndex={disabled || role === 'presentation' ? undefined : -1}
```

Comment:
Looks hacky but maybe ok until we complete this https://github.com/iTwin/iTwinUI-react/issues/349